mac/keyrace-mac/KeyTap.swift

```diff
@@ -47,37 +52,103 @@ func myCGEventCallback(proxy: CGEventTapProxy, type: CGEventType, event: CGEvent
             keyTap.increment(char)
         }
         DispatchQueue.main.async {
-            keyTap.appDelegate.menubarItem?.statusBarItem.button?.title = formatCount(count: keyTap.keycount)
+            keyTap.appDelegate.statusBarItem.button?.attributedTitle = formatCount(count: keyTap.keycount)
         }
     }
 
     if [.tapDisabledByTimeout].contains(type) {
-        keyTap.appDelegate.menubarItem?.statusBarItem.button?.title = "Lost event tap!"
+        keyTap.appDelegate.statusBarItem.button?.title = "Lost event tap!"
     }
 
     return Unmanaged.passRetained(event)
 }
 
-struct Player: Codable {
-    var username: String
-    var gravatar: String
-    var score: Int
-}
-
-class KeyTap {
+// This is an ObservableObject so that the UI can subscribe to updates in players,
+// and we’ll use @Published to send updates whenever the player list changes.
+class KeyTap: ObservableObject {
     var appDelegate : AppDelegate
-    var keycount = 0
     var lastDay = -1
-    var lastMin = -1
     var timerRunning = false
     var keyTrapSetup = false
-    var KEYRACE_HOST = "keyrace.app"
-    var minutes = [Int](repeating:0, count:1440)
-    var keys = [Int](repeating: 0, count:256)
-    var leaderboardText = NSMutableAttributedString()
-
+    var KEYRACE_HOST = "https://keyrace.app"
+    
+    // Load values from UserDefaults.
+    @Published var keycount: Int = UserDefaults.standard.keyCount {
```

Comment:
Removing this `@Published` attribute prevented the app from making all of my typing really slow after I've opened the menu. @ashtom and I looked and couldn't figure out why this would be needed; any issue you see with removing it?

Reply:
UI Might be slower to update but honestly it probably gets it from the @Published on keys anyways

Reply:
will update, sorry for the random ping @published 🤦

Reply:
the only thing its used for in the UI is calculating the percentages in the heatmap

Reply:
Ah ha! Could you reproduce the slowness on your computer? For some reason it happens consistently for me but not @ashtom.